discovery-provider/src/api/v1/tracks.py

```diff
@@ -284,41 +271,41 @@ def get(self):
 
 @full_ns.route("/trending/underground")
 class FullUndergroundTrending(Resource):
-    def get_cache_key(self):
-        """Construct a cache key from user"""
-        request_items = to_dict(request.args)
-        request_items.pop('limit', None)
-        request_items.pop('offset', None)
-        key = extract_key(request.path, request_items.items())
-        return key
-
     @record_metrics
     @full_ns.marshal_with(full_tracks_response)
     def get(self):
         args = underground_trending_parser.parse_args()
-        offset, limit = format_offset(args), format_limit(args, TRENDING_LIMIT)
-        current_user_id = args.get("user_id")
-        args = {
-            'limit': limit,
-            'offset': offset
-        }
+        strategy = trending_selector.get_strategy(TrendingType.UNDERGROUND_TRACKS)
+        trending_tracks = underground_trending(request, args, strategy)
+        return success_response(trending_tracks)
 
-        # If user ID, let get_underground_trending
-        # handle caching + limit + offset
-        if current_user_id:
-            decoded = decode_string_id(current_user_id)
-            args["current_user_id"] = decoded
-            trending = get_underground_trending(args)
-        else:
-            # If no user ID, fetch all cached tracks
-            # and perform pagination here, passing
-            # no args so we get the full list of tracks.
-            key = self.get_cache_key()
-            trending = use_redis_cache(
-                key, TRENDING_TTL_SEC, lambda: get_underground_trending({}))
-            trending = trending[offset: limit + offset]
-        return success_response(trending)
+@full_ns.route("/trending/underground/<string:version>")
+class FullUndergroundTrendingAlternative(Resource):
```

Comment:
Same comment for these guys - wonder about consolidating to the original endpoints.